pgm_learner/DiscreteNodeState[] node_states
